packages/ui/src/components/drawer.tsx

```diff
@@ -0,0 +1,94 @@
+import * as React from 'react'
+
+import { usePortal } from '@/context'
+import { cn } from '@utils/cn'
+import { Drawer as DrawerPrimitive } from 'vaul'
+
+const Drawer = ({
+  shouldScaleBackground = true,
+  nested = false,
+  ...props
+}: React.ComponentProps<typeof DrawerPrimitive.Root>) =>
+  nested ? (
+    <DrawerPrimitive.NestedRoot shouldScaleBackground={shouldScaleBackground} {...props} />
+  ) : (
+    <DrawerPrimitive.Root shouldScaleBackground={shouldScaleBackground} {...props} />
+  )
+Drawer.displayName = 'Drawer'
```

Comment:
To prevent the sprawl of a bazillion individual components, we're trying to use namespacing to keep them contained. We can name this one `DrawerRoot` and then export a `Drawer` namespace at the end instead of individual components.
```suggestion
const DrawerRoot = ({
  shouldScaleBackground = true,
  nested = false,
  ...props
}: React.ComponentProps<typeof DrawerPrimitive.Root>) =>
  nested ? (
    <DrawerPrimitive.NestedRoot shouldScaleBackground={shouldScaleBackground} {...props} />
  ) : (
    <DrawerPrimitive.Root shouldScaleBackground={shouldScaleBackground} {...props} />
  )
DrawerRoot.displayName = 'DrawerRoot'
```

Reply:
agreed, i was planning to make this change later.. but makes sense to do it in the first cut itself. wil do.